test/libsolidity/SemanticTest.cpp

```diff
@@ -192,7 +225,19 @@ vector<SideEffectHook> SemanticTest::makeSideEffectHooks() const
 				return result;
 			}
 			return {};
-		}};
+		},
+		[this](FunctionCall const& _call) -> vector<string>
+		{
+			if (boost::starts_with(_call.signature, "constructor("))
+			{
+				if (ExecutionFramework::transactionSuccessful())
+					m_deployedBytecode = ExecutionFramework::returndata();
+				else
+					m_deployedBytecode.reset();
+			}
```

Comment:
Shouldn't it be reset also when it's not a constructor?

```suggestion
			}
			else
				m_deployedBytecode.reset();
```